File: kotlin/PlanesAndroid/app/src/main/java/com/planes/common/Orientation.java
package com.planes.common;

//plane orientation
public enum Orientation {
    NorthSouth(0),
    SouthNorth(1),
    WestEast(2),
    EastWest(3);

    private final int value;

    Orientation(final int newValue) {
        value = newValue;
    }

    public int getValue() {
        return value;
    }
}

File: kotlin/PlanesCompose/app/src/main/java/com/planes/singleplayerengine/Orientation.kt
package com.planes.singleplayerengine

//plane orientation
enum class Orientation(val value: Int) {
    NorthSouth(0), SouthNorth(1), WestEast(2), EastWest(3);

    companion object {
        fun fromInt(value: Int) = values().first { it.value == value }
    }
}